src/main/java/com/vmware/vipclient/i18n/VIPCfgFactory.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2019-2020 VMware, Inc.
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package com.vmware.vipclient.i18n;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class VIPCfgFactory {
+
+    /**
+     * A lazy-loaded singleton VIPCfg using the "initialization-on-demand holder" design pattern
+     */
+    private static class GlobalCfgHolder {
+        static final VIPCfg globalCfg = new VIPCfg();
```

Comment:
Suggest to save main product's name here, and save the real instance into the 'configs' map. The initial name can be "default".

Reply:
I have deprecated the capability to instantiate with no product name. See new code changes.